pkg/asset/rhcos/image.go

```diff
@@ -52,101 +55,107 @@ func (i *Image) Dependencies() []asset.Asset {
 func (i *Image) Generate(ctx context.Context, p asset.Parents) error {
 	if oi, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); ok && oi != "" {
 		logrus.Warn("Found override for OS Image. Please be warned, this is not advised")
-		*i = Image(oi)
+		*i = *MakeAsset(oi)
 		return nil
 	}
 
 	ic := &installconfig.InstallConfig{}
 	p.Get(ic)
 	config := ic.Config
-	osimage, err := osImage(ctx, config)
+	osimageControlPlane, osImageCompute, err := osImage(ctx, config)
 	if err != nil {
 		return err
 	}
-	*i = Image(osimage)
+	// For compatibility with before the asset had multi-arch support
+	if len(osImageCompute) == 0 {
```

Comment:
It would be nice if this logic is hidden inside osImage().

Reply:
Sure

Reply:
Done